### ISSUE TITLE
Error when linting with `--cache` key

### ISSUE THREAD

**Issue by downace** [intlify/eslint-plugin-vue-i18n#666]
**Tell us about your environment**

- **ESLint version:** 9.38.0
- **eslint-plugin-vue version:** 10.5.1
- **eslint-plugin-vue-i18n version:** 4.1.0
- **Node version:** 20.19.5

**Please show your full configuration:**

```js
import js from '@eslint/js';
import { defineConfigWithVueTs } from '@vue/eslint-config-typescript';
import vueI18n from '@intlify/eslint-plugin-vue-i18n';

export default defineConfigWithVueTs([
  js.configs.recommended,
  ...vueI18n.configs.recommended, // Works ok without this line
  {
    files: ['**/*.vue'],
    languageOptions: {
      ecmaVersion: 'latest',
      sourceType: 'module',
    },

    rules: {},
  },
]);
```

**What did you do?**

Run `eslint --cache`

**What did you expect to happen?**

ESLint should lint my project files

**What actually happened?**

ESLint fails with error:

```
Oops! Something went wrong! :(

ESLint: 9.38.0

The requested operation requires ESLint to serialize configuration data,
but the configuration key "parser.parseForESLint" contains a function value,
which cannot be serialized.

This typically happens when you're using a custom parser that does not
provide a "meta" property, which is how ESLint determines the serialized
representation. Please open an issue with the maintainer of the custom parser
and share this link:

https://eslint.org/docs/latest/extend/custom-parsers#meta-data-in-custom-parsers

If you still have problems, please stop by https://eslint.org/chat/help to chat
with the team.
```

It works without errors if I remove `vueI18n.configs.recommended` config, or run eslint without `--cache` key.

Similar issue with next.js: https://github.com/eslint/eslint/issues/19322
